crates/api_common/src/utils.rs

```diff
@@ -144,6 +145,58 @@ pub fn is_top_mod(
   }
 }
 
+/// Checks to make sure the acting moderator is higher than the target moderator.
+pub async fn check_is_higher_mod(
+  pool: &mut DbPool<'_>,
+  local_user_view: &LocalUserView,
+  community_id: CommunityId,
+  target_person_ids: &[PersonId],
+) -> LemmyResult<()> {
+  CommunityModerator::is_higher_mod_check(
+    pool,
+    community_id,
+    local_user_view.person.id,
+    target_person_ids,
+  )
+  .await
+  .with_lemmy_type(LemmyErrorType::NotHigherMod)?;
+
+  Ok(())
+}
+
+/// Checks to make sure the acting admin is higher than the target admin.
+/// This needs to be done on admin removals, and all purge functions
+pub async fn check_is_higher_admin(
+  pool: &mut DbPool<'_>,
+  local_user_view: &LocalUserView,
+  target_person_ids: &[PersonId],
+) -> LemmyResult<()> {
+  LocalUser::is_higher_admin_check(pool, local_user_view.person.id, target_person_ids)
+    .await
+    .with_lemmy_type(LemmyErrorType::NotHigherAdmin)?;
+
+  Ok(())
+}
+
+/// Checks to make sure the acting admin is higher than the target admin.
+/// This needs to be done on admin removals, and all purge functions
```

Comment:
this comment seems incorrect, purges aren't available to moderators

Reply:
Only the word admin is used here.

Reply:
Ah gotcha, thx.